### PR TITLE
[flink][bug] Fix CDC Ingestion mysql DATETIME type cast error

### LINKED ISSUE #860: [Bug]  CDC Ingestion   mysql  DATETIME(6) cast  error.
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

0.4-SNAPSHOT

### Compute Engine

flink 1.16  
paimon 0.1-SNAPSHOT  
mysql 8

### Minimal reproduce step

      MySqlSyncTableAction mySqlSyncTableAction = (MySqlSyncTableAction) action.get();
            StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
            env.setParallelism(4);
            env.enableCheckpointing(10000);
            env.setStateBackend(new HashMapStateBackend());

            mySqlSyncTableAction.build(env);
            env.execute(String.format("MySQL-Paimon Table Sync: %s.%s","flink","cdc_test"));

### What doesn't meet your expectations?

java.time.format.DateTimeParseException: Text '1649549400000000' could not be parsed at index 0
	at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1777)
	at org.apache.paimon.utils.DateTimeUtils.parseTimestampData(DateTimeUtils.java:382)
	at org.apache.paimon.utils.TypeUtils.toTimestamp(TypeUtils.java:166)
	at org.apache.paimon.utils.TypeUtils.castFromString(TypeUtils.java:117)
	at org.apache.paimon.flink.sink.cdc.CdcRecord.toGenericRow(CdcRecord.java:119)
	at org.apache.paimon.flink.sink.cdc.SchemaAwareStoreWriteOperator.processRecord(SchemaAwareStoreWriteOperator.java:74)
	at org.apache.paimon.flink.sink.cdc.SchemaAwareStoreWriteOperator.processRecord(SchemaAwareStoreWriteOperator.java:45)
	at org.apache.paimon.flink.sink.AbstractStoreWriteOperator.processElement(AbstractStoreWriteOperator.java:123)

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by yuzelin**
### Purpose
Fix #860 

fix DATETIME type cast logic.

### Tests

`MySqlSyncTableActionITCase#testAllTypes`

### API and Format 

No

### Documentation

No

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/test/resources/mysql/setup.sql [hunk 1]
```diff
@@ -95,20 +115,39 @@ CREATE TABLE all_types_table (
...
     12345.11, 12345.22, 12345.33,
+    -- FIXED
     123456789876543212345678987654321.11, 123456789876543212345678987654321.22, 123456789876543212345678987654321.33,
+    -- DECIMAL
     11111, 22222, 33333,
-    '2023-03-23', '2023-03-23 14:30:05', '2023-03-23 15:00:10.123456',
+    -- DATE
+    '2023-03-23',
+    -- DATETIME
+    '2023-03-23 14:30:05', '2023-03-23 14:30:05.123', '2023-03-23 14:30:05.123456',
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/test/resources/mysql/setup.sql:145]: Add data `'2023-03-23 14:30'`

**Comment by yuzelin** [comment]
354466C8-74F2-4798-93B9-ED1A7FCB5A7A

**Review by JingsongLi — APPROVED** [review]
+1

**Comment by zhengshubin** [comment]
mysql  type TIMESTAMP also make error : Failed to convert value 2023-04-01 00:10 to type TIMESTAMP(6). Waiting for schema update.
java.time.format.DateTimeParseException: Text '2023-04-01 00:10' could not be parsed, unparsed text found at index 10

**Comment by yuzelin** [comment]
@zhengshubin Thanks for reporting. I will try to fix it.

**Comment by yuzelin** [comment]
@zhengshubin Hi, fixed: https://github.com/apache/incubator-paimon/pull/1014

**Pull request merged**